apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/BigtableClientTest.cs

```diff
@@ -293,6 +292,57 @@ private class TestBigtableClient : BigtableClient { }
                 () => client.ReadRowAsync(tableName, rowKey, filter));
         }
 
+        [Fact]
+        public async Task ReadRows_Family_Order()
+        {
+            var stream = new MockReadRowsStream(
+                new ReadRowsResponse
+                {
+                    Chunks =
+                    {
+                        new ReadRowsResponse.Types.CellChunk
+                        {
+                            RowKey = ByteString.CopyFromUtf8("abc"),
+                            FamilyName = "z",
+                            Qualifier = ByteString.CopyFromUtf8("column1"),
+                            Value = ByteString.CopyFromUtf8("value1")
+                        },
+                        new ReadRowsResponse.Types.CellChunk
+                        {
+                            RowKey = ByteString.CopyFromUtf8("abc"),
+                            FamilyName = "A",
+                            Qualifier = ByteString.CopyFromUtf8("column2"),
+                            Value = ByteString.CopyFromUtf8("value2")
+                        }
+                    }
+                },
+                new ReadRowsResponse
+                {
+                    Chunks =
+                    {
+                        new ReadRowsResponse.Types.CellChunk
+                        {
+                            RowKey = ByteString.CopyFromUtf8("abc"),
+                            FamilyName = "a",
+                            Qualifier = ByteString.CopyFromUtf8("column3"),
+                            Value = ByteString.CopyFromUtf8("value3"),
+                            CommitRow = true
+                        }
+                    }
+                });
+
+            int rowCount = 0;
+            await stream.AsAsyncEnumerable().ForEachAsync(row =>
+            {
+                rowCount++;
+                var familyNames = row.Families.Select(r => r.Name).ToArray();
+                Assert.Equal("A", familyNames[0]);
+                Assert.Equal("a", familyNames[1]);
+                Assert.Equal("z", familyNames[2]);
+            });
+            Assert.Equal(1, rowCount);
```

Comment:
Add an assertion that each family has the right value associated with it as well?

Reply:
Done. Also added a 'Z' family to verify we're really getting ordinal and not invariant sorting.

Reply:
Aha. I never know what case-sensitive but invariant sorting does with things like this. Thanks for adding the test :)